R/summary.pop_data.R

```diff
@@ -21,122 +21,67 @@ summary.pop_data <- function(object, strata = NULL, ...) {
   value_column <- object %>% get_value_var()
   id_column <- object %>% get_id_var()
 
-  if (is.null(strata)) {
-    ages <-
-      object %>%
-      distinct(
-        .data[[age_column]],
-        .data[[id_column]]
-      )
+  # create a list of the columns
+  cols <- c(age_column, id_column, strata)
 
-    age_summary <-
-      ages %>%
-      pull(age_column) %>%
-      summary() %>%
-      print()
-
-    cat("\nn =", nrow(ages), "\n")
-
-    cat("\nDistribution of age: \n\n")
-
-    ab_summary <-
-      object %>%
-      dplyr::summarize(
-        .by = .data$antigen_iso,
-        Min = object %>%
-          get_value() %>%
-          min(na.rm = TRUE),
-        `1st Qu.` = object %>%
-          get_value() %>%
-          quantile(.25, na.rm = TRUE),
-        Median = object %>%
-          get_value() %>%
-          median(),
-        `3rd Qu.` = object %>%
-          get_value() %>%
-          quantile(.75, na.rm = TRUE),
-        Max = object %>%
-          get_value() %>%
-          max(na.rm = TRUE),
-        `# NAs` = object %>%
-          get_value() %>%
-          is.na() %>%
-          sum()
-      ) %>%
-      as.data.frame() %>%
-      print()
-
-    to_return <- list(
-      n = nrow(ages),
-      age_summary = age_summary,
-      ab_summary = ab_summary
+  ages <-
+    object %>%
+    distinct(
+      across(all_of(cols))
     )
 
-    return(invisible(to_return))
-  } else {
-    ages <-
-      object %>%
-      distinct(
-        .data[[age_column]],
-        .data[[id_column]],
-        .data[[strata]]
-      )
-
-    cat("\nn =", nrow(ages), "\n")
+  age_summary <-
+    ages %>%
+    select(
+      all_of(c(age_column, strata))
+    ) %>%
+    summarise(
+      age_min = min(.data[[age_column]]),
+      age_first_quartile = quantile(.data[[age_column]], 0.25),
+      age_median = median(.data[[age_column]]),
+      age_mean = mean(.data[[age_column]]),
+      age_third_quartile = quantile(.data[[age_column]], 0.75),
+      age_max = max(.data[[age_column]]),
+      .by = strata
+    ) %>%
+    print()
 
-    cat("\nDistribution of age: \n\n")
-    age_summary <-
-      ages %>%
-      select(
-        all_of(age_column),
-        all_of(strata)
-      ) %>%
-      summarise(
-        age_min = min(.data[[age_column]]),
-        age_first_quartile = quantile(.data[[age_column]], 0.25),
-        age_median = median(.data[[age_column]]),
-        age_mean = mean(.data[[age_column]]),
-        age_third_quartile = quantile(.data[[age_column]], 0.75),
-        age_max = max(.data[[age_column]]),
-        .by = strata
-      ) %>%
-      print()
+  cat("\nn =", nrow(ages), "\n")
 
-    cat("\nDistributions of antigen-isotype measurements:\n\n")
+  cat("\nDistribution of age: \n\n")
```

Comment:
lines 49-51 are still out of order relative to lines 33-47, causing the message "Distribution of age:" to appear after the corresponding table:

<img width="858" alt="image" src="https://github.com/UCD-SERG/serocalculator/assets/2474437/53e151be-7aab-4c51-a223-9c72f0619682">


The message is supposed to appear before the first table.

Reply:
also the prefix "age_" in the first table is redundant; the whole table is about age. please remove the prefix from that table.

Reply:
This changes have been made.